watertap/property_models/ion_DSPMDE_prop_pack.py

```diff
@@ -640,21 +790,68 @@ def define_state_vars(self):
                 "temperature": self.temperature,
                 "pressure": self.pressure}
 
-    def assert_electroneutrality(self, tol=None, tee=False):
+    def assert_electroneutrality(self, tol=None, tee=True, defined_state=True, adjust_by_ion=None,
+                                 get_property=None, solve=True):
         if tol is None:
-            tol = 1e-6
+            tol = 0
+        if adjust_by_ion is not None:
+            if adjust_by_ion in self.params.solute_set:
+                self.charge_balance = Constraint(expr=sum(self.charge_comp[j] * self.conc_mol_phase_comp['Liq', j]
+                                                 for j in self.params.solute_set) == 0)
+            else:
+                raise ValueError('adjust_by_ion must be set to the name of an ion in the list of solutes.')
+        if defined_state:
             for j in self.params.solute_set:
-                if not self.flow_mol_phase_comp['Liq', j].is_fixed():
+                if not self.flow_mol_phase_comp['Liq', j].is_fixed() and adjust_by_ion != j:
                     raise AssertionError(
                         f"{self.flow_mol_phase_comp['Liq', j]} was not fixed. Fix flow_mol_phase_comp for each solute"
                         f" to check that electroneutrality is satisfied.")
-        val = value(sum(self.charge_comp[j] * self.flow_mol_phase_comp['Liq', j]
-                    for j in self.params.solute_set))
+                if adjust_by_ion == j and self.flow_mol_phase_comp['Liq', j].is_fixed():
+                    self.flow_mol_phase_comp['Liq', j].unfix()
+        else:
+            for j in self.params.solute_set:
+                if self.flow_mol_phase_comp['Liq', j].is_fixed():
+                    raise AssertionError(
+                        f"{self.flow_mol_phase_comp['Liq', j]} was fixed. Either set defined_state=True or unfix "
+                        f"flow_mol_phase_comp for each solute to check that electroneutrality is satisfied.")
+
+        if get_property is not None:
+            for i in [get_property]:
+                getattr(self, i)
+        self.conc_mol_phase_comp
```

Comment:
We might just comment that we're accessing these properties here to ensure their creation. In a "typical" program this and the `getattr` above would be a no-op.